src/spectrum_feed_server.rs

```diff
@@ -0,0 +1,98 @@
+use std::fmt;
+use std::io::Write;
+use std::net::{TcpListener, TcpStream, ToSocketAddrs};
+
+use flume::Receiver;
+
+#[derive(Debug)]
+pub enum Error {
+    BindSocketFailed(std::io::Error),
```

Comment:
This error type is a bit overkill TBH. I just started out with an error design I'm familiar with and then turned out there was only one possible error. I can probably simplify this to a static string or something?

In a *library* errors should be structured and well defined. But in an application they just need to print prettily unless the application itself need to destructure them for some reason.